diff --color -urN binutils-2.45/bfd/config.bfd binutils-2.45-nuovo/bfd/config.bfd
--- binutils-2.45/bfd/config.bfd	2025-07-27 01:00:00.000000000 +0200
+++ binutils-2.45-nuovo/bfd/config.bfd	2026-09-21 15:58:20.577885721 +0200
@@ -586,7 +586,10 @@
     targ_defvec=hppa_som_vec
     targ_selvecs=hppa_elf32_vec
     ;;
-
+  i[3-7]86-*-miosix*)
+    targ_defvec=i386_elf32_vec
+    targ_selvecs="i386_elf32_vec"
+    ;;
   i[3-7]86-*-elf* | i[3-7]86-*-rtems* | i[3-7]86-*-genode*)
     targ_defvec=i386_elf32_vec
     targ_selvecs="iamcu_elf32_vec i386_coff_vec"
diff --color -urN binutils-2.45/config.sub binutils-2.45-nuovo/config.sub
--- binutils-2.45/config.sub	2025-07-27 01:00:00.000000000 +0200
+++ binutils-2.45-nuovo/config.sub	2026-09-21 15:58:20.580777278 +0200
@@ -1722,6 +1722,9 @@
 			;;
 		esac
 		;;
+	miosix*)
+		os=miosix
+		;;
 	aout* | coff* | elf* | pe*)
 		# These are machine code file formats, not OSes
 		obj=$os
diff --color -urN binutils-2.45/gas/configure.tgt binutils-2.45-nuovo/gas/configure.tgt
--- binutils-2.45/gas/configure.tgt	2025-07-27 01:00:00.000000000 +0200
+++ binutils-2.45-nuovo/gas/configure.tgt	2026-09-21 15:58:20.581016336 +0200
@@ -225,6 +225,7 @@
   h8300-*-elf)				fmt=elf ;;
   h8300-*-linux*)			fmt=elf em=linux ;;
 
+  i386-*-miosix*)           fmt=elf ;;
   i386-*-beospe*)			fmt=coff em=pe ;;
   i386-*-beos*)				fmt=elf ;;
   i386-*-elfiamcu)			fmt=elf arch=iamcu ;;
diff --color -urN binutils-2.45/ld/configure.tgt binutils-2.45-nuovo/ld/configure.tgt
--- binutils-2.45/ld/configure.tgt	2025-07-27 01:00:00.000000000 +0200
+++ binutils-2.45-nuovo/ld/configure.tgt	2026-09-21 15:58:20.581370229 +0200
@@ -399,6 +399,9 @@
 			targ_emul=elf_i386
 			targ_extra_emuls="elf_iamcu"
 			;;
+i[3-7]86-*-miosix*)
+			targ_emul=elf_i386
+			;;
 i[3-7]86-*-elfiamcu)	targ_emul=elf_iamcu
 			targ_extra_emuls=elf_i386
 			;;
