diff --color -urN gcc-15.2.0/config.sub gcc-15.2.0-nuovo/config.sub
--- gcc-15.2.0/config.sub	2025-08-08 08:51:39.866337996 +0200
+++ gcc-15.2.0-nuovo/config.sub	2026-09-21 15:58:28.365734976 +0200
@@ -1732,7 +1732,7 @@
 	     | mirbsd* | netbsd* | dicos* | openedition* | ose* \
 	     | bitrig* | openbsd* | secbsd* | solidbsd* | libertybsd* | os108* \
 	     | ekkobsd* | freebsd* | riscix* | lynxos* | os400* \
-	     | bosx* | nextstep* | cxux* | aout* | elf* | oabi* \
+	     | bosx* | nextstep* | cxux* | aout* | elf* | miosix* | oabi* \
 	     | ptx* | coff* | ecoff* | winnt* | domain* | vsta* \
 	     | udi* | lites* | ieee* | go32* | aux* | hcos* \
 	     | chorusrdb* | cegcc* | glidix* | serenity* \
diff --color -urN gcc-15.2.0/gcc/config/i386/miosix.h gcc-15.2.0-nuovo/gcc/config/i386/miosix.h
--- gcc-15.2.0/gcc/config/i386/miosix.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-15.2.0-nuovo/gcc/config/i386/miosix.h	2026-10-03 11:51:17.090725769 +0200
@@ -0,0 +1,117 @@
+/* Target definitions for the Miosix RTOS. */
+
+/* RATIONALE: adding builtin_define to always define _MIOSIX
+   - when libgcc/libstdc++/newlib are compiled, as there are some #ifdef _MIOSIX
+   - when Miosix processes are compiled, to allow #ifdef _MIOSIX
+   Also add versioning to miosix-specific compiler patches, and specs for
+   making GCC compile Miosix processes by default. */
+
+#undef  TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()                    \
+  do                                                \
+    {                                               \
+      builtin_define ("_MIOSIX");                   \
+      builtin_define ("_MIOSIX_GCC_PATCH_MAJOR=4"); \
+      builtin_define ("_MIOSIX_GCC_PATCH_MINOR=2"); \
+      builtin_assert ("system=miosix");             \
+    }                                               \
+  while (0)
+
+/* Force -fpie -msingle-pic-base for Miosix process object files */
+#undef  OS_CC1_SPEC
+#define OS_CC1_SPEC "%{!qkernelspace:-fpie}"
+
+/* No crti.o/crtbegin.o/crt0.o */
+#undef  STARTFILE_SPEC
+#define STARTFILE_SPEC ""
+
+/* No crtend.o/crtn.o */
+#undef  ENDFILE_SPEC
+#define ENDFILE_SPEC ""
+
+/* Library specs for Miosix processes */
+
+/* STD_LIB_SPEC is the default value of LIB_SPEC as defined in gcc.cc, what it
+   does is to expand to -lc in most cases except if profiling is enabled. */
+#define STD_LIB_SPEC "%{!shared:%{g*:-lg} %{!p:%{!pg:-lc}}%{p:-lc_p}%{pg:-lc_p}}"
+
+/* Includes libsyscalls and libatomic if -qkernelspace is not specified, and
+   the libraries are found.
+   If -lstdc++ is used, move it inside the system LIB_SPEC because in Miosix
+   it has a circular dependency on -lsyscalls. Also -lm must be moved in
+   because -lstdc++ depends on -lm.*/
+/* The case in which the libraries are not found is important to allow the
+   configure steps for GCC's libraries to succeed, because they try to compile
+   linked executables (even though they shouldn't need to!). */
+/* This spec goes unused if -nostlib is passed to gcc (i.e. we don't
+   have to handle that case manually ourselves). Additionally, it is also
+   inside a --start-group --end-group. This is implemented in
+   the parent spec LINK_GCC_C_SEQUENCE_SPEC in unknown-elf.h. */
+/* IMPORTANT: we CANNOT remove the -lstdc++ outfile because LIB_SPEC
+   is evaluated TWICE: one time for creating the --pass-through-libs=...
+   arguments to the linker, the other for adding the libraries to the command
+   line. This is not evident because of too many levels of indirection, but
+   it happens like this:
+     LINK_COMMAND_SPEC
+       string-includes LINK_PLUGIN_SPEC
+         uses %(link_gcc_c_sequence) to invoke LINK_GCC_C_SEQUENCE_SPEC
+           uses %L to invoke LIB_SPEC
+       uses %X to add the output files to the commandline
+       uses %(link_gcc_c_sequence) to invoke LINK_GCC_C_SEQUENCE_SPEC
+         uses %L to invoke LIB_SPEC
+   Unfortunately the output files are added to the linker's command line
+   (by spec command %X) before the second call of LIB_SPEC, so it is basically
+   impossible to remove -lstdc++ once it is added again here.
+   There is no other way to handle libstdc++ because its presence is not
+   defined in any spec string, instead it is hardcoded directly in the
+   C++ driver. */
+#undef  LIB_SPEC
+#define LIB_SPEC \
+  "%{!qkernelspace: \
+     %:if-exists-then-else(libsyscalls.a%s -lsyscalls) \
+     %:if-exists-then-else(libatomic.a%s -latomic) \
+     %{%:find-outfile(-lstdc++):-lstdc++ -lm}} " STD_LIB_SPEC
+
+/* Linker specs for Miosix processes */
+
+/* Always remove -lpthread because its functions are already defined either by
+   the rest of the kernel, or by libsyscalls for processes.
+   Also add required options for processes to the linker's command line if
+   -qkernelspace is not specified. Note that the default LINK_SPEC is empty. */
+#undef  LINK_SPEC
+#define LINK_SPEC \
+  "%:remove-outfile(-lpthread) \
+   %{!qkernelspace:-n -pie --spare-dynamic-tags 3}"
+   
+/* Additional linker options for Miosix processes */
+
+/* If compiling a process executable (no -qkernelspace) and no custom linker
+   script is specified (no -T on command line) add the default linker script.
+   Linker scripts must go at the end of the linker's command line, otherwise
+   they would not be searched in all paths. */
+/* Unfortunately, there is no spec for adding a linker script, so we have to
+   modify the whole LINK_COMMAND_SPEC. This is not even unusual, darwin.h
+   also does it. In the future we might even want to customize this spec
+   even more, but it needs to kept synchronized with the original one in
+   gcc.cc. The modified part is just after %{T*}, the rest is the same. */
+#undef  LINK_COMMAND_SPEC
+#define LINK_COMMAND_SPEC "\
+%{!fsyntax-only:%{!c:%{!M:%{!MM:%{!E:%{!S:\
+    %(linker) " \
+    LINK_PLUGIN_SPEC \
+   "%{flto|flto=*:%<fcompare-debug*} \
+    %{flto} %{fno-lto} %{flto=*} %l " LINK_PIE_SPEC \
+   "%{fuse-ld=*:-fuse-ld=%*} " LINK_COMPRESS_DEBUG_SPEC \
+   "%X %{o*} %{e*} %{N} %{n} %{r}\
+    %{s} %{t} %{u*} %{z} %{Z} %{!nostdlib:%{!r:%{!nostartfiles:%S}}} \
+    %{static|no-pie|static-pie:} %@{L*} %(link_libgcc) " \
+    VTABLE_VERIFICATION_SPEC " " SANITIZER_EARLY_SPEC " %o "" \
+    %{fopenacc|fopenmp|%:gt(%{ftree-parallelize-loops=*:%*} 1):\
+    %:include(libgomp.spec)%(link_gomp)}\
+    %{fgnu-tm:%:include(libitm.spec)%(link_itm)}\
+    " STACK_SPLIT_SPEC "\
+    %{fprofile-arcs|fcondition-coverage|fpath-coverage|fprofile-generate*|coverage:-lgcov} " SANITIZER_SPEC " \
+    %{!nostdlib:%{!r:%{!nodefaultlibs:%(link_ssp) %(link_gcc_c_sequence)}}}\
+    %{!nostdlib:%{!r:%{!nostartfiles:%E}}} \
+    %{T*} %{!qkernelspace:%{!T:-T miosix_process.ld}} \
+    \n%(post_link) }}}}}}"
diff --color -urN gcc-15.2.0/gcc/config/miosix.h gcc-15.2.0-nuovo/gcc/config/miosix.h
--- gcc-15.2.0/gcc/config/miosix.h	2026-10-06 17:12:50.275944331 +0200
+++ gcc-15.2.0-nuovo/gcc/config/miosix.h	2026-10-03 12:03:32.319172212 +0200
@@ -82,7 +82,7 @@
 #undef  LINK_SPEC
 #define LINK_SPEC \
   "%:remove-outfile(-lpthread) \
-   %{!qkernelspace:-n -pie --spare-dynamic-tags 3 --target2=mx-data-rel}"
+   %{!qkernelspace:-n -pie --spare-dynamic-tags 3}"
    
 /* Additional linker options for Miosix processes */
 
diff --color -urN gcc-15.2.0/gcc/config.gcc gcc-15.2.0-nuovo/gcc/config.gcc
--- gcc-15.2.0/gcc/config.gcc	2026-10-06 17:12:50.276481127 +0200
+++ gcc-15.2.0-nuovo/gcc/config.gcc	2026-09-21 15:58:30.441461405 +0200
@@ -1965,6 +1965,11 @@
 i[34567]86-*-elfiamcu)
 	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h newlib-stdint.h i386/iamcu.h"
 	;;
+i[34567]86-*-miosix* | i[34567]86-miosix-*)
+	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h newlib-stdint.h i386/i386elf.h i386/miosix.h"
+	extra_options="${extra_options} miosix.opt"
+	use_gcc_stdint=wrap
+	;;
 i[34567]86-*-elf*)
 	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h newlib-stdint.h i386/i386elf.h"
 	;;
diff --color -urN gcc-15.2.0/libatomic/configure.tgt gcc-15.2.0-nuovo/libatomic/configure.tgt
--- gcc-15.2.0/libatomic/configure.tgt	2026-10-06 17:12:50.282715637 +0200
+++ gcc-15.2.0-nuovo/libatomic/configure.tgt	2026-10-03 17:44:39.178446543 +0200
@@ -167,11 +167,14 @@
             ;;
         esac
 	;;
-
-  arm*-miosix-eabi*)
+  i[3456]86-miosix-elf*)
+	config_path="miosix ${config_path}"
+	;;
+  miosix-eabi*)
 	config_path="miosix"
 	;;
 
+
   *-*-rtems*)
 	XCFLAGS="${configure_tgt_pre_target_cpu_XCFLAGS}"
 	config_path="rtems"
diff --color -urN gcc-15.2.0/libstdc++-v3/include/bits/this_thread_sleep.h gcc-15.2.0-nuovo/libstdc++-v3/include/bits/this_thread_sleep.h
--- gcc-15.2.0/libstdc++-v3/include/bits/this_thread_sleep.h	2026-10-06 17:12:50.291600391 +0200
+++ gcc-15.2.0-nuovo/libstdc++-v3/include/bits/this_thread_sleep.h	2026-09-21 15:58:31.373231261 +0200
@@ -42,6 +42,8 @@
 # include <time.h>  // nanosleep
 #endif
 
+#include <bits/chrono.h>
+# include <cerrno>
 namespace std _GLIBCXX_VISIBILITY(default)
 {
 _GLIBCXX_BEGIN_NAMESPACE_VERSION
